hoomd/md/ComputeThermoGPU.cc

```diff
@@ -182,6 +187,7 @@ void ComputeThermoGPU::computeProperties()
         args.external_energy = m_pdata->getExternalEnergy();
 
         // perform the computation on the GPU(s)
+        m_tuner->begin();
         gpu_compute_thermo_partial(d_properties.data,
```

Comment:
`gpu_compute_thermo_partial` determines the number of partial sums it computes. In order to pass that information on to `gpu_compute_thermo_final`, add an output argument `unsigned int &num_partial_sums` to `gpu_compute_thermo_partial` and pass it to `gpu_compute_thermo_final`.